src/helpers.cpp

```diff
@@ -106,9 +106,9 @@ ada_really_inline bool shorten_path(std::string& path,
   // If url’s scheme is "file", path’s size is 1, and path[0] is a normalized
   // Windows drive letter, then return.
   if (type == ada::scheme::type::FILE &&
-      first_delimiter == std::string_view::npos) {
+      first_delimiter == std::string_view::npos && !path.empty()) {
```

Comment:
We don't need this anymore due to `helpers::substring` safety right?

Reply:
We do, we do, because we later changed how `substring` works. Initially, it would check and return an empty string dynamically, but it no longer does that. Instead we have our of our asserts in place. So we need to ensure that `path` is not empty.

Reply:
I changed that because the check was costing us optimization-wise. So I replaced the dynamic check with an assert.